front/src/modules/pathfinding/utils.ts

```diff
@@ -33,9 +36,38 @@ export const formatSuggestedOperationalPoints = (
     offsetOnTrack: op.part.position,
     track: op.part.track,
     positionOnPath: op.position,
-    coordinates: getPointCoordinates(geometry, pathLength, op.position),
+    coordinates:
+      (geometry !== undefined &&
+        pathLength !== undefined &&
+        getPointCoordinates(geometry, pathLength, op.position)) ||
+      undefined,
   }));
 
+export const addTrackNamesToSuggestedOperationalPoints = async (
+  operationalPoints: SuggestedOP[],
+  getTrackSectionsByIds: (requestedTrackIds: string[]) => Promise<InfraObjectWithGeometry[]>
+): Promise<SuggestedOP[]> => {
+  // Extract track IDs to fetch track names
+  const trackIds = uniq(operationalPoints.map((op) => op.track));
+  const trackSections = await getTrackSectionsByIds(trackIds);
+
+  return operationalPoints.map((op) => {
+    const associatedTrackSection = trackSections.find(
+      (trackSection) => (trackSection.railjson as TrackSection).id === op.track
+    );
+
+    const trackName = associatedTrackSection
+      ? (associatedTrackSection.railjson as TrackSectionEntity['properties']).extensions?.sncf
+          ?.track_name
+      : null;
+
+    return {
+      ...op,
+      trackName: trackName || undefined,
```

Comment:
You can use undefined by default here and remove it l66
```suggestion
      : undefined;

    return {
      ...op,
      trackName: trackName,
```

Reply:
Nice catch! I don't understand why I wrote that in the first place ^^''

Reply:
Actually let me simplify this further